### ISSUE TITLE
Cache replaces manual edits

### ISSUE THREAD

**Issue by zenturacp** [cline/cline#950]
### Which API Provider are you using?

Anthropic

### Which Model are you using?

Claude 3.5 Sonnet 20241022

### What happened?

I see often when i do manual edits in files and then subsequent changes reverts my manual edits, looks like some sort of cache, it can be a different task im working on, and it just reverts back things

### Steps to reproduce

I have a Azure Function in node.js it has 2 methods GET, POST

I remove POST as a method, ask cline to make a util and import it in the Function file, then it imports it and changes back the GET to GET, POST, but it have never been discussed to change methods and the util can be  just a folder name sanitizer etc.

Also seen it in package.json versions just suddenly beeing changed - also depenedencies that have not been touced for several chats and suddenly the project wont start and i diff and see it have changed the version to somthing else

On a day i experience it 3-4 times ish (6-8 hrs), i dont have any current examples but next time i see it i will provide the API output

**Comment by SmileyChris** [comment]
I've found if you yell at cline and tell it you changed the file it'll usually re-read it then apply the changes again correctly.

**Comment by siscia** [comment]
Yes, I found the same! Sometimes it also realizes by itself if it cannot figure out the diff. It usually happens when I ran out of tokens.

**Comment by saoudrizwan** [comment]
> I see often when i do manual edits in files and then subsequent changes reverts my manual edits

> Also seen it in package.json versions just suddenly beeing changed - also depenedencies that have not been touced for several chats and suddenly the project wont start and i diff and see it have changed the version to somthing else

Both cases of the model being stupid and can be resolved with specific prompting, ie in Settings>Custom Instructions asking to read package.json before modifying it. For the manual edits being overwritten, the extension tells the model that the user made these manual edits and to take them into consideration moving forward but I have run into intances where it ignores this. Will need to try to reproduce and see if there some way to modify the tool response to write_to_file/replace_in_file and prevent this from happening.
